nipoppy_cli/tests/test_workflow_pipeline.py

```diff
@@ -588,8 +585,39 @@ def test_run_main_catch_errors(workflow: PipelineWorkflow):
     )
     manifest.save_with_backup(workflow.layout.fpath_manifest)
     workflow.run_main()
-    assert workflow._n_runs == 1
-    assert workflow._n_errors == 1
+    assert workflow.n_total == 1
+    assert workflow.n_success == 0
+    assert workflow.return_code == ReturnCode.ERROR_RUN_SINGLE
+
+
+@pytest.mark.parametrize(
+    "n_success,n_total,expected_message",
+    [
+        (0, 0, "No participant-session pairs to run"),
+        (1, 2, f"[{LogColor.PARTIAL_SUCCESS}]Ran"),
+        (0, 1, f"[{LogColor.FAILURE}]Ran"),
+        (2, 2, f"[{LogColor.SUCCESS}]Successfully ran"),
```

Comment:
Same comment as in file `nipoppy_cli/tests/test_workflow_dicom_reorg.py`, to include full log message.